cmd/cigarToBed/cigarToBed.go

```diff
@@ -0,0 +1,198 @@
+		curr = genomeGraph.AddNode(answer, curr)
+		if aln[i].Op == 0 {
+			genomeGraph.AddEdge(&answer.Nodes[i-1], curr, 1)
+			genomeGraph.AddEdge(&answer.Nodes[i-2], curr, 0.5)
+		} else if aln[i].Op == 1 || aln[i].Op == 2 {
+			genomeGraph.AddEdge(&answer.Nodes[i-1], curr, 0.5)
+		} else {
+			log.Fatalf("Error: cigar.Op = %d is unrecognized...\n", aln[i].Op)
+		}
+	}
+	return answer
+}
+
+//raven moved helper functions from main and non-usage functions into this function
+func GlobalAlignment_CigarToBed(inputFileOne *fileio.EasyReader, inputFileTwo *fileio.EasyReader, outFa string, outIns_bed string, outDel_bed string, FirstPos_InsBed int, FirstPos_DelBed int, Chrom string) {
+	//make sure files meet the usage requirements of globalAlignment.go
+	faOne, faDoneOne := fasta.NextFasta(inputFileOne)
+	faTwo, faDoneTwo := fasta.NextFasta(inputFileTwo)
+	fasta.ToUpper(faOne) //raven added these 2 lines to prevent lowercase letters from indexing out of range in align/linearGap.go
+	fasta.ToUpper(faTwo)
+	if faDoneOne || faDoneTwo {
+		log.Fatalf("error, unable to read .fa files, check for > symbol before each name and that each fasta entry is only one line. Check to make sure there are only four bases or N, globalAlignment is unable to use anything outside A-T-G-C-N.\n")
+	}
+	numSeqOne := CountSeqIdx(inputFileOne)
+	numSeqTwo := CountSeqIdx(inputFileTwo)
+	if numSeqOne > 1 || numSeqTwo > 1 {
+		log.Fatalf("multiple sequnces detected in .fa files: %v sequences in the first .fa file and %v sequences in the second .fa file. This program is designed for .fa files with only 1 sequence in them\n", numSeqOne, numSeqTwo)
+	}
+	//attempt at Dan's len(faOne) suggestion
+	//if len(string(faOne)) != 1 || len(string(faTwo)) != 1 {
+	//log.Fatalf("multiple sequnces detected in .fa files. This program is designed for .fa files with only 1 sequence in them.\n")
+	//}
+
+	//needleman wunsch (global alignment)
+	//raven's note: scoring matrices are in align/align.go, e.g. HumanChimp matrix, same as what UCSC Genome Browser uses
+	//raven's note: ConstGap is a function in align/linearGap.go, with only constant gapPen (gap penalty)
+	//bestScore, aln := align.ConstGap(faOne.Seq, faTwo.Seq, align.HumanChimpTwoScoreMatrix, -430)
+	//fmt.Printf("Alignment score is %d, cigar is %v \n", bestScore, aln)
+	//raven's edit: try AffineGap in align/affineGap.go, with diffrent gapOpen and gapExtend penalty, since gapExtend<gapOpen, encourages big gap in order to align better in unbroken chunks
+	bestScore, aln := align.AffineGap(faOne.Seq, faTwo.Seq, align.HumanChimpTwoScoreMatrix, -600, -150)
+	fmt.Printf("Using AffineGap, Alignment score is %d, cigar is %v \n", bestScore, aln)
+
+	//TODO: raven's cigarToBed main body starts here
+	//type definitions: Cigar is defined in align/align.go; aln is a []cigar, according to ConstGap defined in align/linearGap.go
+	//modeled after: mafIndels.go, wigPeaks.go
+	//insertion bed
+	insBed := fileio.EasyCreate(outIns_bed)
+	defer insBed.Close()
```

Comment:
Best to avoid defer close when working with the fileio package so you can check the error. Because the fileio writers are buffered and potentially gzipped, there can be some serious errors.

Reply:
Just fixed this. Thank you!